socorro/unittest/processor/test_general_transform_rules.py

```diff
@@ -99,6 +101,61 @@
 })
 
 
+class TestDeNullRule(object):
+    @pytest.mark.parametrize('data, expected', [
+        # no nulls--just making sure things are good
+        ('abc', 'abc'),
+        (b'abc', b'abc'),
+        (123, 123),
+
+        # has nulls
+        ('abc\u0000', 'abc'),
+        ('abc\0', 'abc'),
+        ('ab\0c\0', 'abc'),
+        (b'abc\0', b'abc'),
+        (b'a\0bc\0', b'abc'),
+    ])
+    def test_de_null(self, data, expected):
+        config = get_basic_config()
+        rule = DeNullRule(config)
+        assert rule.de_null(data) == expected
+
+    def test_rule_with_dict(self):
+        raw_crash = {
+            'key1': 'val1',
+            b'\0key2': b'val2\0',
+            '\0key3': '\0val3'
+        }
+
+        config = get_basic_config()
+        rule = DeNullRule(config)
+        rule.act(raw_crash, {}, {}, get_basic_processor_meta())
+
+        assert raw_crash == {
+            'key1': 'val1',
+            b'key2': b'val2',
+            'key3': 'val3'
+        }
+
+    def test_rule_with_dotdict(self):
+        # NOTE(willkg): DotDict doesn't like bytes keys
+        raw_crash = DotDict({
+            'key1': 'val1',
+            '\0key2': b'val2\0',
+            '\0key3': '\0val3'
+        })
+
+        config = get_basic_config()
+        rule = DeNullRule(config)
+        rule.act(raw_crash, {}, {}, get_basic_processor_meta())
+
+        assert raw_crash == DotDict({
+            'key1': 'val1',
+            'key2': b'val2',
+            'key3': 'val3'
+        })
```

Comment:
Currently, the raw crash is a DotDict. When we do a processor rewrite, we'll change that. So it's important to test both possibilities: dict and DotDict.

Reply:
Hrm... that reminds me that the raw crash is read in by boto crashstorage and at that point it's tossed into a DotDict. If it has null keys then, that'll fail. I'll look into that.

Reply:
Nevermind--this is fine. JSON can't store binary data so we're guaranteed for the keys to always be string types.